info-compiler/src/main/java/com/google/sps/infocompiler/InfoCompiler.java

```diff
@@ -270,28 +274,39 @@ void storeElectionContestInDatabase(String electionQueryId, JsonObject contest)
     datastore.update(electionEntity);
   }
 
-  private boolean checkIncumbency(String candidateName, String position, String division) {
+  private boolean checkIncumbency(Map<String, String> map, String position, String name) {
+    for (Map.Entry<String, String> e : map.entrySet()) {
+      if (e.getKey() == position && e.getValue() == name) return true;
+    }
+    return false;
```

Comment:
Isn't the map keyed by position? Utilize the map interface: `return map.get(position).equals(name);`  Or if you end up switching it to `Map<String, List<String>>`: `return map.get(position).contains(name);`.  You can also probably inline this logic in `storeElectionContestCandidateInDatabase()` at this point.